Source/Core/ClippingPlaneCollection.js

```diff
@@ -126,34 +167,91 @@ define([
+         * @type {Boolean}
+         * @default false
+         */
+        enabled : {
+            get : function() {
+                return this._enabled;
+            },
+            set : function(value) {
+                if (this._enabled === value) {
+                    return;
+                }
+                this._enabled = value;
+            }
+        },
+
+        /**
+         * Returns a texture containing packed, untransformed clipping planes.
+         *
+         * @memberof ClippingPlaneCollection.prototype
+         * @type {Texture}
+         * @readonly
+         * @private
+         */
+        texture : {
+            get : function() {
+                return this._clippingPlanesTexture;
+            }
+        },
+
+        /**
+         * A reference to the ClippingPlaneCollection's owner, if any.
+         *
+         * @memberof ClippingPlaneCollection.prototype
+         * @readonly
+         * @private
+         */
+        owner : {
+            get : function() {
+                return this._owner;
             }
         }
     });
 
+    function setIndexDirty(collection, index) {
+        // If there's already a different _dirtyIndex set, more than one plane has changed since update.
+        // Entire texture must be reloaded
+        collection._manyDirtyPlanes = collection._manyDirtyPlanes || (collection._dirtyIndex !== -1 && collection._dirtyIndex !== index);
+        collection._dirtyIndex = index;
+    }
+
     /**
-     * Adds the specified {@link Plane} to the collection to be used to selectively disable rendering
+     * Adds the specified {@link ClippingPlane} to the collection to be used to selectively disable rendering
      * on the outside of each plane. Use {@link ClippingPlaneCollection#unionClippingRegions} to modify
      * how modify the clipping behavior of multiple planes.
      *
-     * @param {Plane} plane The plane to add to the collection.
-     *
-     * @exception {DeveloperError} The plane added exceeds the maximum number of supported clipping planes.
+     * @param {ClippingPlane} plane The ClippingPlane to add to the collection.
      *
      * @see ClippingPlaneCollection#unionClippingRegions
      * @see ClippingPlaneCollection#remove
      * @see ClippingPlaneCollection#removeAll
      */
     ClippingPlaneCollection.prototype.add = function(plane) {
-        //>>includeStart('debug', pragmas.debug);
-        if (this.length >= ClippingPlaneCollection.MAX_CLIPPING_PLANES) {
-            throw new DeveloperError('The maximum number of clipping planes supported is ' + ClippingPlaneCollection.MAX_CLIPPING_PLANES);
+        var newPlaneIndex = this._planes.length;
+        if (plane instanceof ClippingPlane) {
+            var that = this;
+            plane.onChangeCallback = function(index) {
+                setIndexDirty(that, index);
+            };
+            plane.index = newPlaneIndex;
+        } else {
+            this._containsUntrackablePlanes = true;
```

Comment:
This should print a `deprecationWarning` too

Reply:
more use cases for a "whoops" reaction for github...